### PR TITLE
[xgboostserver] Convert list input to numpy array before creating DMatrix

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:
Due to https://github.com/dmlc/xgboost/pull/3970 , prediction calculated by current xgboost model is incorrect for some cases. The result deviates quite significantly compared to training set. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fix incorrect prediction result in xgboost server.

**Special notes for your reviewer**:

- Additional dependency (numpy) is added to xgboost server

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by pradithya** [comment]
It seems the CI failure is unrelated to the PR

```
        testset = torchvision.datasets.CIFAR10(root='./data', train=False,
>                                              download=True, transform=transform)
```

File: python/xgbserver/setup.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
     python_requires='>3.4',
     packages=find_packages("xgbserver"),
     install_requires=[
+        "numpy>=1.0"
```
Comment (yuzisun) [inline python/xgbserver/setup.py:36]: kfserving package already include the numpy dependency

Comment (pradithya) [reply python/xgbserver/setup.py]: Updated in new commit

File: python/xgbserver/xgbserver/model.py [hunk 2]
```diff
@@ -43,7 +44,7 @@ def load(self) -> bool:
     def predict(self, request: Dict) -> Dict:
         try:
             # Use of list as input is deprecated see https://github.com/dmlc/xgboost/pull/3970
```
Comment (pradithya) [inline python/xgbserver/xgbserver/model.py:46]: @yuzisun I am curious why this hasn't been fixed before even though there is a comment pointing to the issue in xgboost.

Comment (yuzisun) [reply python/xgbserver/xgbserver/model.py:46]: @pradithya We are mostly moving to v2 protocol at this point for xgboost and I think it is fixed there https://github.com/SeldonIO/MLServer/blob/739ebbaf92f1c135c789df009f2c386c6ab6f49e/runtimes/xgboost/mlserver_xgboost/xgboost.py#L49.

**Comment by yuzisun** [comment]
@pradithya If you want this to be released, please help cherry pick onto the release-0.5 branch.

File: python/xgbserver/xgbserver/model.py (hunk 2 above)
Comment (pradithya) [reply python/xgbserver/xgbserver/model.py:46]: Got it, thanks for clarifying @yuzisun

**Pull request merged**